File: 26. Remove Duplicates from Sorted Array.cpp
// 26. Remove Duplicates from Sorted Array
// Time: O(N)
// Space: O(1)

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() <= 1)
            return nums.size();
        int index = 1;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] != nums[index - 1])
                nums[index++] = nums[i];
        }
        return index;
    }
};

File: 26. Remove Duplicates from Sorted Array.py
# 26. Remove Duplicates from Sorted Array
# Time: O(N)
# Space: O(1)

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 1:
            return len(nums);
        
        idx = 0
        for i in range(1, len(nums)):
            if nums[i - 1] != nums[i]:
                idx += 1
                nums[idx] = nums[i]
        return idx + 1
